### PR TITLE
Improve voltage computation for lumped ports

### PR DESCRIPTION
**Pull request by dmarek-flex**
The E field is discontinuous at the port boundaries where conductors are present. In this picture the ground plane is at z=0 and the strip is at z=1000. Interpolation is inaccurate there, and even though only the end points are affected, it has a noticeable effect on accuracy.

image

Now, all field points outside the port bounds are ignored and linear extrapolation is used to estimate the E field on the exact port bounds.

### CODE REVIEW

File: tidy3d/plugins/smatrix/component_modelers/terminal.py [hunk 1]
```diff
@@ -193,16 +194,25 @@ def port_voltage(port: LumpedPort, sim_data: SimulationData) -> xr.DataArray:
             e_component = "xyz"[port.voltage_axis]
             field_data = sim_data[f"{port.name}_E{e_component}"]
             e_field = field_data.field_components[f"E{e_component}"]
+
+            # Remove E field outside the port region
+            e_field = e_field.sel(
+                {
+                    e_component: slice(
+                        port.bounds[0][port.voltage_axis], port.bounds[1][port.voltage_axis]
+                    )
```
Comment (weiliangjin2021) [inline tidy3d/plugins/smatrix/component_modelers/terminal.py:203]: When you use `slice` here, is it exclusive on both start and end? My impression is that it is inclusive in the beginning.

**Review by tylerflex — APPROVED** [review]
Looks good to me, was going to say to include a changelog item but it probably isn't necessary since the original version of this hasn't been released yet, right?

**Comment by dmarek-flex** [comment]
> Looks good to me, was going to say to include a changelog item but it probably isn't necessary since the original version of this hasn't been released yet, right?

Yea I was about to add something to the changelog and saw the lumped ports stuff right above, so I decided not to change anything.

File: tidy3d/plugins/smatrix/component_modelers/terminal.py (hunk 1 above)
Comment (dmarek-flex) [reply tidy3d/plugins/smatrix/component_modelers/terminal.py]: Now I understand your point, I added which makes the excludes the boundaries. I didn't see a better way so I used `drop_sel`

**Review by weiliangjin2021 — APPROVED** [review]

**Pull request merged**